docker/client_test.go

```diff
@@ -0,0 +1,23 @@
+package main
+
+import (
+	"os"
+	"testing"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/utils"
+)
+
+func TestClientDebugEnabled(t *testing.T) {
+	defer utils.DisableDebug()
+
+	clientFlags.Common.FlagSet.Parse([]string{"-D"})
+	clientFlags.PostParse()
+
+	if os.Getenv("DEBUG") != "1" {
+		t.Fatal("expected debug enabled, got false")
+	}
+	if logrus.GetLevel() != logrus.DebugLevel {
+		t.Fatalf("expected logrus debug level, got %v", logrus.GetLevel())
+	}
+}
```

Comment:
I wasn't looking to diff the exact output from the cli debug msgs, rather just something with the word "[DEBU]" it in or to make sure "docker -D info" shows debug info ( ala https://github.com/docker/docker/issues/16264#issuecomment-178250407 ). Just something to prove it get past the flag parsing logic and into the rest of the code.